src/Confluent.Kafka/ProducerBuilder.cs

```diff
@@ -20,6 +20,39 @@
 
 namespace Confluent.Kafka
 {
+    /// <summary>
+    ///     Calculate a partition number given a <paramref name="topic" />,
```

Comment:
The topic name is typically not relevant in keying.

Reply:
it's there and in the docs because it's part of the partitioner api. it could be relevant if someone is setting a default partitioner... but i can't think of a specific use case. i will add a note that it is not typically relevant.

Reply:
maybe this is where you were coming from with the perf. question.

if it's slow (don't reckon it'll be too material), we can add another type in the future `SimplePartitionerDelegate` or something, but i think for now the more general option is the one to go with.